native/acter/src/api/account/three_pid.rs

```diff
@@ -93,27 +155,81 @@ impl ThreePidManager {
                 };
                 account.set_account_data(content).await?;
 
+                Ok(ThreePidEmailTokenResponse {
+                    inner,
+                    client_secret,
+                })
+            })
+            .await?
+    }
+
+    // add 3pid on the homeserver for this account
+    // this 3pid may be used by the homeserver to authenticate the user during sensitive operations
+    #[cfg(feature = "testing")]
+    pub async fn add_3pid(
+        &self,
+        client_secret: String,
+        sid: String,
+        password: String,
+    ) -> Result<bool> {
+        let client = self.client.deref().clone();
+        let account = self.account.clone();
+        let user_id = self.user_id.clone();
+        let client_secret = ClientSecret::parse(&client_secret)?;
+        let sid = SessionId::parse(&sid)?; // it was already related with email or msisdn
+
+        RUNTIME
+            .spawn(async move {
+                let capabilities = client.get_capabilities().await?;
+                if !capabilities.thirdparty_id_changes.enabled {
+                    bail!("Server doesn't support 3pid change");
+                }
+                if let Err(e) = account.add_3pid(&client_secret, &sid, None).await {
+                    let Some(inf) = e.as_uiaa_response() else {
+                        return Err(clearify_error(e));
+                    };
+                    let pswd = assign!(Password::new(user_id.into(), password), {
+                        session: inf.session.clone(),
+                    });
+                    let auth_data = AuthData::Password(pswd);
+                    account
+                        .add_3pid(&client_secret, &sid, Some(auth_data))
+                        .await
+                        .map_err(clearify_error)?;
+                }
                 Ok(true)
             })
             .await?
     }
 
+    pub async fn external_ids(&self) -> Result<Vec<ExternalId>> {
+        let account = self.account.clone();
+
+        RUNTIME
+            .spawn(async move {
+                let resp = account.get_3pids().await.map_err(clearify_error)?;
+                let records = resp.threepids.iter().map(ExternalId::new).collect();
```

Comment:
inline with what we are doing above, we need to change this to:
 ```suggestion
                let records = resp.threepids.into_iter().map(ExternalId::new).collect();
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1896/commits/09694374f6d71f9613089967aa77338fff99ed91